javaharness/java/arcs/android/impl/ArcsLocal.java

```diff
@@ -1,102 +0,0 @@
-package arcs.android.impl;
-
-import javax.inject.Inject;
-
-import arcs.api.ArcData;
-import arcs.api.Arcs;
-import arcs.api.ArcsEnvironment;
-import arcs.api.PECInnerPort;
-import arcs.api.PecPortManager;
-import arcs.api.PortableJson;
-import arcs.api.PortableJsonParser;
-import arcs.api.UiBroker;
-
-// This class implements Arcs API for callers within the same Android service
-// that hosts the Arcs Runtime.
-public class ArcsLocal implements Arcs {
-
-  private final ArcsEnvironment environment;
-  private final PecPortManager pecPortManager;
-  private final PortableJsonParser jsonParser;
-  private final UiBroker uiBroker;
-
-  @Inject
-  ArcsLocal(
-      ArcsEnvironment environment,
-      PecPortManager pecPortManager,
-      PortableJsonParser jsonParser,
-      UiBroker uiBroker) {
-    this.environment = environment;
-    this.pecPortManager = pecPortManager;
-    this.jsonParser = jsonParser;
-    this.uiBroker = uiBroker;
-  }
-
-  @Override
-  public void runArc(ArcData arcData) {
-    PECInnerPort pecInnerPort = null;
-    for (ArcData.ParticleData particleData : arcData.getParticleList()) {
-      if (particleData.getParticle() != null) {
-        if (pecInnerPort == null) {
-          pecInnerPort =
-              pecPortManager.getOrCreateInnerPort(arcData.getPecId(), arcData.getSessionId());
-        }
-        pecInnerPort.mapParticle(particleData.getParticle());
-      }
-    }
-    environment.sendMessageToArcs(constructRunArcRequest(arcData));
-  }
-
-  @Override
-  public void stopArc(ArcData arcData) {
-    environment.sendMessageToArcs(constructStopArcRequest(arcData));
```

Comment:
copied from comment in original pr:

This class was useful for cases when you wanted to run an Arc from within the Arcs service (this is how the "ingestion-arc" would be started, it just wasn't part of the prototype, but it is part of the design).
I think it's nice if you could use the same interface to run an Arc from any service - Arcs or other Android service, this is why I created the Arcs interface and ArcsLocal & ArcsAndroid implementations (the names might be changed, but is there a reason to get rid of ArcsLocal implementation as a concept?)